app/src/main/java/com/example/photogalery/VisibleFragment.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.photogalery
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.util.Log
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+
+abstract class VisibleFragment : Fragment() {
```

Comment:
You can retain the `@LayoutRes int contentLayoutId` param in the constructor and pass it to the child Fragment.

Reply:
The name of this class implies that this is only available when the fragment is visible. That is not true. This class will be seen in the hierarchy even when it is not in foreground.
For a better name, we have to look at what it always does. That is dismissing notifications.
I suggest using `PollNotificationHandlerFragment`. If you have a better name, you can use that